app-center-services/src/main/java/org/exoplatform/appcenter/service/ApplicationCenterService.java

```diff
@@ -449,7 +449,11 @@ public GeneralSettings getAppGeneralSettings() throws Exception { // NOSONAR
    */
   public ApplicationList getApplicationsList(int offset, int limit, String keyword) {
     ApplicationList applicationList = new ApplicationList();
-    List<Application> applications = appCenterStorage.getApplications(keyword, offset, limit);
+    List<Application> applications = appCenterStorage.getApplications(keyword);
+    if (limit <= 0) {
+      limit = applications.size();
+    }
+    applications = applications.stream().skip(offset).limit(limit).collect(Collectors.toList());
     applicationList.setApplications(applications);
     long totalApplications = appCenterStorage.countApplications();
     applicationList.setSize(totalApplications);
```

Comment:
This size should be the size of the filtered list.
Currently, if 3 applications on 10 match the keyword, the size is 10. But it should be 3, else the paginator will not be able to display well.
So for me, we should have long totalApplications=applications.size(); and this should be done before the stream usage.

Reply:
Okay will fix it